lib/splitclient-rb/clients/split_client.rb

```diff
@@ -241,5 +171,112 @@ def valid_client
       end
       SplitIoClient.configuration.valid_mode
     end
+
+    def treatments(key, split_names, attributes = {}, metric = 'sdk.get_treatments')
+      return nil unless SplitIoClient::Validators.valid_get_treatments_parameters(split_names)
+
+      sanitized_split_names = sanitize_split_names(split_names)
+
+      if sanitized_split_names.empty?
+        SplitIoClient.configuration.logger.error('get_treatments: split_names must be a non-empty Array')
+        return {}
+      end
+
+      bucketing_key, matching_key = keys_from_key(key)
+      bucketing_key = bucketing_key ? bucketing_key.to_s : nil
+      matching_key = matching_key ? matching_key.to_s : nil
+
+      evaluator = Engine::Parser::Evaluator.new(@segments_repository, @splits_repository, true)
+      start = Time.now
+      treatments_labels_change_numbers =
+        @splits_repository.get_splits(sanitized_split_names).each_with_object({}) do |(name, data), memo|
+          memo.merge!(name => treatment(key, name, attributes, data, false, true, evaluator))
+        end
+      latency = (Time.now - start) * 1000.0
+      # Measure
+      @adapter.metrics.time(metric, latency)
+
+      unless SplitIoClient.configuration.disable_impressions
+        time = (Time.now.to_f * 1000.0).to_i
+        @impressions_repository.add_bulk(
+          matching_key, bucketing_key, treatments_labels_change_numbers, time
+        )
+
+        route_impressions(sanitized_split_names, matching_key, bucketing_key, time, treatments_labels_change_numbers, attributes)
+      end
+
+      split_names_keys = treatments_labels_change_numbers.keys
+      treatments = treatments_labels_change_numbers.values.map do |v|
+        {
+          treatment: v[:treatment],
+          configurations: v[:configurations]
```

Comment:
Won't mark all appearances but this one changes too, to config. Sorry for the late notice.